include/jemalloc/internal/arena_structs_b.h

```diff
@@ -122,14 +122,18 @@ struct arena_bin_s {
 	extent_t		*slabcur;
 
 	/*
-	 * Heap of non-full slabs.  This heap is used to assure that new
-	 * allocations come from the non-full slab that is oldest/lowest in
+	 * Heaps of non-empty, non-full slabs.  This heap is used to ensure that
+	 * new allocations come from the non-full slab that is oldest/lowest in
 	 * memory.
+	 * When picking an extent to allocate from, we prefer slabs from the
+	 * sized alloc region, but always prefer active slabs to non-active
+	 * ones.
 	 */
-	extent_heap_t		slabs_nonfull;
+	extent_heap_t		sized_alloc_region_nonfull_slabs;
+	extent_heap_t		other_nonfull_slabs;
 
-	/* List used to track full slabs. */
-	extent_list_t		slabs_full;
+	/* List used to track full slabs (regardless of their origin). */
+	extent_list_t		full_slabs;
```

Comment:
Yeah, sounds reasonable. I originally changed them because, once there were four of them, having it be adjective-noun made it feel more readable. Once we're used to it I suppose it doesn't matter.